src/proxyImplementationTemplate.ts

```diff
@@ -0,0 +1,27 @@
+export function proxyImplementationTemplate({
+  creatorFnName,
+  ownValueAlias,
+  useTranslateFn,
+}: {
+  creatorFnName: string;
+  ownValueAlias: string;
+  useTranslateFn: boolean;
+}) {
+  return `const ${creatorFnName} = <R extends string>(t: (...args: unknown[]) => R = (k) => k as R, prevKeys: string = ''): unknown => 
+  new Proxy(() => {}, {
+    get: (_, key: string) => {
+      if (key === '${ownValueAlias}') {
+        return ${creatorFnName}(t, prevKeys)
+      }
+
+      if (prevKeys) {
+        return ${creatorFnName}(t, [prevKeys, key].join('.'));
+      }
+
+      return ${creatorFnName}(t, key);
+    },
+    apply: (_, __, argArray) => {
+      return ${useTranslateFn ? `t(prevKeys, ...argArray)` : 'prevKeys'};
+    },
+  });`;
+}
```

Comment:
This is exactly what I've did recently with data-hooks in settings 🤯 

https://github.com/wix-private/wixstores-settings/blob/bfcm-2024/packages/wixstores-client-settings-editor/src/gallery/services/dataHooks/dataHooks.ts

Reply:
Same 😄 .. I just wanted to make it support nesting, and fit our dataHooks approach